### PR TITLE
Build server file structure

### PR DESCRIPTION
**Pull request by haitham-akram**
relates #7

### CODE REVIEW

**Review by AbdallahAbujazar — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -4,7 +4,11 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "start": "cross-env NODE_ENV=production node dist/index.js",
+    "dev": "cross-env NODE_ENV=development ts-node-dev src/index.ts",
+    "build": "tsc -p .",
+    "postbuild": "npm run build && cd client/ && npm i && npm run build",
```
Comment (Mu7ammadAbed) [inline package.json:10]: this script makes a problem when we run the `build` script, so maybe we could change it's name to `deployment-postbuild`
```suggestion
    "deployment-postbuild": "npm run build && cd client/ && npm i && npm run build",
```

File: src/app.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import express from 'express'
+import compression from 'compression'
+import cookieParser from 'cookie-parser'
+import dotenv from 'dotenv'
+import router from './routes'
+
+dotenv.config()
```
Comment (Mu7ammadAbed) [inline src/app.ts:7]:
```suggestion
dotenv.config()
const { NODE_ENV, PORT } = process.env;
```

File: src/app.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const app = express()
+app.disabled('x-powered-by')
+app.set('port', process.env.PORT ?? 3000)
+app.use(express.json())
+app.use(express.urlencoded({ extended: false }))
+app.use(compression())
+app.use(cookieParser())
+app.use(router)
```
Comment (Mu7ammadAbed) [inline src/app.ts:15]:
```suggestion
app.use('/api/v1', router);

if (NODE_ENV === 'production') {
  app.use(express.static(join(__dirname, '..', 'client', 'build')));

  app.get('/*', (req, res) => {
    res.sendFile(join(__dirname, '..', 'client', 'build', 'index.html'));
  });
}
```

Comment (Mu7ammadAbed) [reply src/app.ts:15]: and try to organize the file a little bit more by adding empty line between the middleware and the router for example, and move the `app.set` to the end just before the router

File: src/app.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import cookieParser from 'cookie-parser'
+import dotenv from 'dotenv'
+import { join } from 'path'
+import router from './routes'
+
+dotenv.config()
+const { NODE_ENV, PORT } = process.env
+const app = express()
+
+app.disabled('x-powered-by')
```
Comment (Mu7ammadAbed) [inline src/app.ts:12]:
```suggestion
app.disable('x-powered-by')
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by MohammedSallout — APPROVED** [review]

**Pull request merged**